### PR TITLE
Delete coalesceSR, coalesceEJ, coalesceTJ

### PR DESCRIPTION
**Pull request by alissapajer**
Depends on #3919

### CODE REVIEW

File: qscript/src/main/scala/quasar/qscript/rewrites/Rewrite.scala [hunk 1]
```diff
@@ -108,199 +101,35 @@ class Rewrite[T[_[_]]: BirecursiveT: EqualT: ShowT: RenderTreeT] extends TTypes[
...
-      liftFFTrans[QScriptCore, G, T[G]](qcPrism)(compactQC(_: QScriptCore[T[G]])),
-      liftFGTrans[QScriptCore, G, T[G]](qcPrism)(compactLeftShift[G](qcPrism)),
-      liftFFTrans[QScriptCore, G, T[G]](qcPrism)(uniqueBuckets(_: QScriptCore[T[G]])),
-      liftFFTrans[QScriptCore, G, T[G]](qcPrism)(compactReductions(_: QScriptCore[T[G]])),
-      liftFFTrans[F, G, T[G]](prism)(C.coalesceQC[G](prism)),
-      liftFGTrans[F, G, T[G]](prism)(normalizeJoins),
-      liftFGTrans[QScriptCore, G, T[G]](qcPrism)(elideNopQC[G])
-    ))(prism(ftf))
+    ftf => repeatedly[G[T[G]]](
+      liftFFTrans[F, G, T[G]](prism)(C.coalesceQC[G](prism))
```
Comment (alissapajer) [inline qscript/src/main/scala/quasar/qscript/rewrites/Rewrite.scala]: Just one thing left in here! :confetti_ball:

**Comment by jsantos17** [comment]
From what I saw yesterday, I figured we can also delete a large part of `coalesceQC` in `Coalesce[QScriptCore]` [here](https://github.com/slamdata/quasar/blob/410f08334e879b55240ea5cc043bca58fe488b20/qscript/src/main/scala/quasar/qscript/rewrites/Coalesce.scala#L244).

We mostly need to keep the `LeftShift(Reduce(...), ...)`  rewrite since this rewrite removes `UnshiftMap`. The other rewrites are not doing much and `it`s are green without them..

**Comment by alissapajer** [comment]
@jsantos17 agreed, that's exactly what I've already done locally :) PR incoming shortly.

**Comment by alissapajer** [comment]
@jsantos17 actually don't merge this PR. The `InterpretedRead` rewrite isn't getting hit :(

**Comment by alissapajer** [comment]
We need to add a test suite to make sure we generate `InterpretedRead` in cases when we should be. Else it will be all too easy to accidentally introduce regressions.

**Pull request closed without merging**